### PR TITLE
Windows Device Trust documentation

### PR DESCRIPTION
**Pull request by strideynet**
Modifies the existing device trust documentation to provide the alternate steps for setting up a Windows TPM based machine instead.

Closes https://github.com/gravitational/teleport.e/issues/1594

### CODE REVIEW

**Comment by strideynet** [comment]
Do not merge until the feature flag is removed.

File: docs/pages/access-controls/guides/device-trust.mdx
Comment (strideynet) [inline docs/pages/access-controls/guides/device-trust.mdx:83]: Revisit this version number before merging.

File: docs/pages/access-controls/guides/device-trust.mdx
Comment (zmb3) [inline docs/pages/access-controls/guides/device-trust.mdx:31]:
```suggestion
Device Trust leverages dedicated hardware in devices to store sensitive
```

File: docs/pages/access-controls/guides/device-trust.mdx
Comment (zmb3) [inline docs/pages/access-controls/guides/device-trust.mdx:185]: Was `tsh device collect` also added to the CLI reference doc?

**Review by zmb3 — APPROVED** [review]

File: docs/pages/access-controls/guides/device-trust.mdx
Comment (strideynet) [reply docs/pages/access-controls/guides/device-trust.mdx:185]: Mm, not sure, this wasn't added as part of the TPM work.

Comment (codingllama) [reply docs/pages/access-controls/guides/device-trust.mdx:185]: I have mentioned hidden commands in docs before, like the "diag" commands, without necessarily adding them to the CLI reference. My 2c is that I wouldn't add this one, but your call.

File: docs/pages/access-controls/guides/device-trust.mdx
Comment (codingllama) [inline docs/pages/access-controls/guides/device-trust.mdx:31]:
```suggestion
Device Trust leverages dedicated secure hardware in devices to store sensitive
```

Just to be extra clear.

File: docs/pages/access-controls/guides/device-trust.mdx
Comment (codingllama) [reply docs/pages/access-controls/guides/device-trust.mdx:83]: Would TPM support, by definition, bump the version to 13.2?

File: docs/pages/access-controls/guides/device-trust.mdx
Comment (codingllama) [reply docs/pages/access-controls/guides/device-trust.mdx:185]: We could also expose a simpler, `tsh device serial-number` command, that spits out the serial number and nothing else. Then we could write a single section here.

File: docs/pages/access-controls/guides/device-trust.mdx
Comment (codingllama) [inline docs/pages/access-controls/guides/device-trust.mdx:199]:
```suggestion
$ export OS=macos
$ export SERIAL='CHANGEME'
$ tctl devices add --os="$OS" --asset-tag="$SERIAL"
```

SERIAL could be `export SERIAL="$(tsh device serial-number)"`, per comment above.

File: docs/pages/access-controls/guides/device-trust.mdx
Comment (codingllama) [inline docs/pages/access-controls/guides/device-trust.mdx:289]: Suggestion: Since this is necessarily v13+, I would cut all the blurb and go straight for `tctl edit`.

**Review by codingllama — APPROVED** [review]
LGTM.

File: docs/pages/access-controls/guides/device-trust.mdx
Comment (strideynet) [reply docs/pages/access-controls/guides/device-trust.mdx:185]: I like the creation of `tsh device serial-number` command - and it'll be easy enough for us to do. I suggest we merge these docs as is, and then do that command as a separate exercise though.

File: docs/pages/access-controls/guides/device-trust.mdx
Comment (strideynet) [reply docs/pages/access-controls/guides/device-trust.mdx:83]: It'll be going out in the next patch release, rather than waiting for a minor release as per conversations with @zmb3

File: docs/pages/access-controls/guides/device-trust.mdx
Comment (strideynet) [reply docs/pages/access-controls/guides/device-trust.mdx:182]: It's a little awkward because of the fallback behaviour. We could list all three powershell commands, I'm a little less sure about the UI location for some of these. I wasn't able to find the system asset tag 😓

File: docs/pages/access-controls/guides/device-trust.mdx
Comment (strideynet) [reply docs/pages/access-controls/guides/device-trust.mdx:199]: It feels a little confusing to have commands in the same code block that need to be run on the admin machine and also on the machine to be enrolled.

File: docs/pages/access-controls/guides/device-trust.mdx
Comment (strideynet) [reply docs/pages/access-controls/guides/device-trust.mdx:289]: Agreed

File: docs/pages/access-controls/guides/device-trust.mdx
Comment (codingllama) [reply docs/pages/access-controls/guides/device-trust.mdx:185]: Sounds good to me.

File: docs/pages/access-controls/guides/device-trust.mdx
Comment (codingllama) [reply docs/pages/access-controls/guides/device-trust.mdx:182]: `tsh device serial-number` makes this moot, so no worries.

File: docs/pages/access-controls/guides/device-trust.mdx
Comment (strideynet) [reply docs/pages/access-controls/guides/device-trust.mdx]: 420fd9a1af

File: docs/pages/access-controls/guides/device-trust.mdx
Comment (strideynet) [reply docs/pages/access-controls/guides/device-trust.mdx]: 420fd9a1af

File: docs/pages/access-controls/guides/device-trust.mdx
Comment (lsgunn-teleport) [inline docs/pages/access-controls/guides/device-trust.mdx:38]:
```suggestion
On Windows devices, a Trusted Platform Module (TPM) is used to perform an
```
it's more common to put the full name before the acronym.

File: docs/pages/access-controls/guides/device-trust.mdx
Comment (lsgunn-teleport) [inline docs/pages/access-controls/guides/device-trust.mdx:40]:
```suggestion
private key that is also protected by the TPM. The signed attestation ensures that the Teleport Auth Server knows both
```

File: docs/pages/access-controls/guides/device-trust.mdx
Comment (lsgunn-teleport) [inline docs/pages/access-controls/guides/device-trust.mdx:41]:
```suggestion
the state of the device and that the request has come from
```

File: docs/pages/access-controls/guides/device-trust.mdx
Comment (lsgunn-teleport) [inline docs/pages/access-controls/guides/device-trust.mdx:75]:
```suggestion
- To enroll a macOS device, you need:
```

File: docs/pages/access-controls/guides/device-trust.mdx
Comment (lsgunn-teleport) [inline docs/pages/access-controls/guides/device-trust.mdx:79]:
```suggestion
- To enroll a Windows device, you need:
```

File: docs/pages/access-controls/guides/device-trust.mdx
Comment (lsgunn-teleport) [inline docs/pages/access-controls/guides/device-trust.mdx:78]:
```suggestion
  - A Teleport version newer than v12.0.0.
  
```

File: docs/pages/access-controls/guides/device-trust.mdx
Comment (lsgunn-teleport) [inline docs/pages/access-controls/guides/device-trust.mdx:155]:
```suggestion
Before you can enroll the device, you need to register it. To register
```
I see that Teleport docs use a lot of first person plural (we). It's a little unusual except in tutorials. My suggestion is more "typical" but feel free to ignore if the point of view feels right to you.

File: docs/pages/access-controls/guides/device-trust.mdx
Comment (lsgunn-teleport) [inline docs/pages/access-controls/guides/device-trust.mdx:156]:
```suggestion
a device, you first need to determine its serial number.
```
just a suggestion to remove the split infinitive.

File: docs/pages/access-controls/guides/device-trust.mdx
Comment (lsgunn-teleport) [inline docs/pages/access-controls/guides/device-trust.mdx:195]:
```suggestion
Replace the `$SERIAL` variable below with the serial number obtained from
```

File: docs/pages/access-controls/guides/device-trust.mdx
Comment (lsgunn-teleport) [inline docs/pages/access-controls/guides/device-trust.mdx:197]:
```suggestion
the following command:
```

**Review by codingllama — APPROVED** [review]

File: docs/pages/access-controls/guides/device-trust.mdx
Comment (lsgunn-teleport) [inline docs/pages/access-controls/guides/device-trust.mdx:226]:
```suggestion
Enrollments are tied to the current user in the device. If the same device is
```
maybe something like:
Device enrollment is specifically linked to the user who performed the enrollment. If the same device is handed to a new user, that user must enroll the device again.

File: docs/pages/access-controls/guides/device-trust.mdx
Comment (lsgunn-teleport) [inline docs/pages/access-controls/guides/device-trust.mdx:249]:
```suggestion
On Windows, the `tsh device enroll` command attempts to elevate to
```

File: docs/pages/access-controls/guides/device-trust.mdx
Comment (lsgunn-teleport) [inline docs/pages/access-controls/guides/device-trust.mdx:250]:
```suggestion
administrator privileges on the device. Privilege elevation is required to perform a
```

File: docs/pages/access-controls/guides/device-trust.mdx
Comment (lsgunn-teleport) [inline docs/pages/access-controls/guides/device-trust.mdx:251]:
```suggestion
TPM Credential Activation. This step is only required for enrollment and is not
```

File: docs/pages/access-controls/guides/device-trust.mdx
Comment (lsgunn-teleport) [inline docs/pages/access-controls/guides/device-trust.mdx:262]:
```suggestion
Some TPMs include a certificate—known as an EKCert—signed by the
```

File: docs/pages/access-controls/guides/device-trust.mdx
Comment (lsgunn-teleport) [inline docs/pages/access-controls/guides/device-trust.mdx:263]:
```suggestion
manufacturer's certificate authority (CA). This certificate allows a third party
```

File: docs/pages/access-controls/guides/device-trust.mdx
Comment (lsgunn-teleport) [inline docs/pages/access-controls/guides/device-trust.mdx:274]:
```suggestion
CA specified in the field will be able to enroll. Previously enrolled devices
```
```suggestion
CA specified in the field will be able to enroll. Previously-enrolled devices
```

File: docs/pages/access-controls/guides/device-trust.mdx
Comment (lsgunn-teleport) [inline docs/pages/access-controls/guides/device-trust.mdx:281]:
```suggestion
After you obtain the CA certificate in PEM format, there are two ways of
```

File: docs/pages/access-controls/guides/device-trust.mdx
Comment (lsgunn-teleport) [inline docs/pages/access-controls/guides/device-trust.mdx:286]:
```suggestion
  for clusters in a highly available configuration.
```

File: docs/pages/access-controls/guides/device-trust.mdx
Comment (lsgunn-teleport) [inline docs/pages/access-controls/guides/device-trust.mdx:288]:
```suggestion
  clusters and is the most flexible.
```

**Review by lsgunn-teleport — APPROVED** [review]
I made some minor "writing convention" type of suggestions, but they are mostly stylistic or simplifications. One repeated word typo. Other than that, I learneda lot.

If any of the suggestions don't make sense or you'd like clarification, just let me know.

File: docs/pages/access-controls/guides/device-trust.mdx
Comment (strideynet) [reply docs/pages/access-controls/guides/device-trust.mdx:155]: Definitely prefer "you" here, thanks for pointing this out!

File: docs/pages/access-controls/guides/device-trust.mdx
Comment (strideynet) [reply docs/pages/access-controls/guides/device-trust.mdx:227]: Yeah, that's nicer. Thanks.

**Pull request merged**
